src/main/scala/com/github/tkawachi/doctest/ScaladocExtractor.scala

```diff
@@ -12,8 +15,25 @@ object ScaladocExtractor {
     Set(DocToken.CodeBlock, DocToken.Description, DocToken.Example)
 
   def extract(scalaSource: String): List[ScaladocComment] = {
-
-    val code = scalaSource.parse[Source].get
+    extractFromInput(Input.String(scalaSource))
+  }
+  def extractFromFile(file: Path, encoding: String): List[ScaladocComment] = {
+    val text = new String(Files.readAllBytes(file), encoding)
+    // Workaround from https://github.com/scalameta/scalameta/issues/443#issuecomment-314797969
+    val trimmedText = text.replace("\r\n", "\n")
+    val input = Input.VirtualFile(file.toString, trimmedText)
+    extractFromInput(input)
+  }
+  def extractFromInput(scalaSource: Input): List[ScaladocComment] = {
+    scalaSource.parse[Source] match {
+      case Parsed.Success(tree) =>
+        extractFromTree(tree)
+      case error @ Parsed.Error(_, _, _) =>
+        println(error.toString())
```

Comment:
This results in `sbt test` passing when there are parse errors so we might want a better solution. I personally recommend having a `Reporter` class that can `.error(pos, message)` and also test for `.hasError`. I've been meaning to expose such functionality in Scalameta itself.

Reply:
if parser can't handle it, doesn't it mean that compilation for that file fails? 
if so, then we don't even get to tests and as such to doctests

Reply:
Correct, the compilation would fail in that case.